examples/fda_testing.py

```diff
@@ -7,62 +7,68 @@
+        )  # save volume as a set of sequential images, fda_filename_[1...N].png
+    else:
+        print("There is no oct volume in fda file.")
 
     fundus_image = (
         fda.read_fundus_image()
     )  # returns a  Fundus image with additional metadata if available
-    fundus_image.save(
-        f"{output_dir}/{filepath.split('/')[-1].split('.')[0]}/{filepath.split('/')[-1].split('.')[0]}_fundus.jpg"
-    )
+    if fundus_image is not None:
+        fundus_image.save(
+            f"{output_dir}/{filepath.split('/')[-1].split('.')[0]}/{filepath.split('/')[-1].split('.')[0]}_fundus.jpg"
+        )
+    else:
+        print("There is no fundus image data in fda file.")
     # save Fundus image to output directory, fda_filename_fundus.jpg
 
     fundus_grayscale_image = (
         fda.read_fundus_image_gray_scale()
     )  # returns a  Fundus image with additional metadata if available
-    fundus_grayscale_image.save(
-        f"{output_dir}/{filepath.split('/')[-1].split('.')[0]}/{filepath.split('/')[-1].split('.')[0]}_grayscale_fundus.jpg"
-    )
+    if fundus_grayscale_image is not None:
+        fundus_grayscale_image.save(
+            f"{output_dir}/{filepath.split('/')[-1].split('.')[0]}/{filepath.split('/')[-1].split('.')[0]}_grayscale_fundus.jpg"
+        )
+    else:
+        print("There is no grayscale fundus image data in fda file.")
     # save Fundus image to output directory, fda_filename_grayscale_fundus.jpg
 
     # preparing json for other datas in fda file
+    chunk_dict = fda.get_list_of_file_chunks(show=False)
     output_json = dict()
-    output_json["fda_file_info"] = fda.read_fda_file_info()
-    output_json["hardware_info"] = fda.read_hardware_info()
-    output_json["patient_info"] = fda.read_patient_info()
-    output_json["capture_info"] = fda.read_capture_info()
-    output_json["param_scan"] = fda.read_param_scan()
-    output_json["param_obs"] = fda.read_param_obs()
-    output_json["img_mot_comp"] = fda.read_img_mot_comp()
-    output_json["img_projection"] = fda.read_img_projection()
-    output_json["regist_info"] = fda.read_regist_info()
-    output_json["contour_info"] = fda.read_contour_info()
-    output_json["effective_scan_range"] = fda.read_effective_scan_range()
-    output_json["cornea_curve_result"] = fda.read_cornea_curve_result_info()
-    output_json["cornea_thickness"] = fda.read_result_cornea_thickness()
+    for key in chunk_dict.keys():
+        if key in [
+            b"@IMG_JPEG",
+            b"@IMG_FUNDUS",
+            b"@IMG_TRC_02",
+        ]:  # this chunks are image chunks and extract with another methods
+            continue
+        json_key = key.decode().split("@")[-1].lower()
+        try:
+            output_json[json_key] = fda.read_any_info_and_make_dict(key)
+        except KeyError:
+            print(f"{key} there is no method for getting info from this chunk.")
 
     info_json = json.dumps(output_json)

```

Comment:
the output json file is much more readable with `json.dumps(output_json, indent=4)`

Reply:
Okay adding this to dumps part.